service/controller/resource/drainerfinalizer/resource.go

```diff
@@ -204,6 +211,15 @@ func (r *Resource) ensure(ctx context.Context, obj interface{}) error {
 			}
 		}
 
+		// Timeout drainer configs should simply be deleted without draining since
+		// the time window for draining did expire at this point.
+		for _, dc := range timeoutDrainerConfigs {
+			err = r.deleteDrainerConfig(ctx, dc)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+		}
```

Comment:
I thought this makes sense but would like to hear about more opinions. Now due to the error matcher issues the `DrainerConfig` CRs are not deleted even though they timed out and the instances are gone. Not sure if this may have unforeseen side effects.

Reply:
In Azure we actually restart draining when it times out. So far it has worked for us quite well and currently our only problems are with occasional stuck `coredns` and `nginx-ingress-controller` PODs.

Reply:
In the case here we would start over forever which renders the timeout quite useless. That is why I thought we should draw a line an some point. Do you think this here is wrong or not so good?

Reply:
> Do you think this here is wrong or not so good?

I think that in general this is the right approach. For different circumstances and different customer requirements different approaches are better, but when you can behave accordingly and draw a strict line to timeout, I think that is good.

TL;DR; When no customer is screaming, I like this approach. :+1: